### ISSUE TITLE
DataTable: RowToggler refactoring

### ISSUE THREAD

**Issue by Rapster** [primefaces/primefaces#6384]
**Description**
Just like #6372, I think the row toggling feature could be enabled more easily. Configuration is done on two levels, through `DataTable#expandedRow` and rowToggler component itself.

**Describe the solution you would like**
A simpler way of doing this could be:
```xml
<p:dataTable var="car" value="#{dtBasicView.cars}">

      <p:column style="width:16px">
          <p:rowToggler expanded="#{car.color eq 'black'}" />
      </p:column>

    <p:column headerText="Id">
        <h:outputText value="#{car.id}" />
    </p:column>

    <p:column headerText="Year">
        <h:outputText value="#{car.year}" />
    </p:column>

    <p:rowExpansion>
        <p:panelGrid  columns="2" columnClasses="label,value" style="width:300px">
            <f:facet name="header">
                <p:graphicImage name="demo/images/car/#{car.brand}-big.gif"/>
            </f:facet>
            <h:outputText value="Id:" />
            <h:outputText value="#{car.id}" />
            <h:outputText value="Year" />
            <h:outputText value="#{car.year}" />
            <h:outputText value="Color:" />
            <h:outputText value="#{car.color}" style="color:#{car.color}"/>
            <h:outputText value="Price" />
            <h:outputText value="$#{car.price}" />
        </p:panelGrid>
    </p:rowExpansion>
</p:dataTable>
```

**Additional context**
~The way I see it, rowTogger could inherit from `p:column`, WDYT?~ _(not a good idea since it forces users to use rowtoggler as a column while they don't want to)_

Basically, the change is to move `DataTable#expandedRow` attribute on `RowToggler` itself, WDYT? It would save `RowToggler#getParentTable` call

**Comment by melloware** [comment]
I like it

**Comment by Rapster** [comment]
Since `ui-expanded-row` class is set on `tr`, it's not time efficient to check the presence of `rowToggler `component inside `column` with our(s) current algo(s) to determine whether or not this CSS class should be set. Closing it for now
